sentry-spring-boot-starter/src/main/java/io/sentry/spring/boot/SentryUserHttpServletRequestProcessor.java

```diff
@@ -0,0 +1,43 @@
+package io.sentry.spring.boot;
+
+import com.jakewharton.nopen.annotation.Open;
+import io.sentry.core.EventProcessor;
+import io.sentry.core.SentryEvent;
+import io.sentry.core.protocol.User;
+import javax.servlet.http.HttpServletRequest;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.web.context.request.RequestAttributes;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+/** Attaches user information from the HTTP request to {@link SentryEvent}. */
+@Open
+public class SentryUserHttpServletRequestProcessor implements EventProcessor {
+
+  @Override
+  public SentryEvent process(final @NotNull SentryEvent event, final @Nullable Object hint) {
+    final RequestAttributes requestAttributes = RequestContextHolder.currentRequestAttributes();
+    if (requestAttributes instanceof ServletRequestAttributes) {
+      final HttpServletRequest request =
+          ((ServletRequestAttributes) requestAttributes).getRequest();
+      event.setUser(toUser(event.getUser(), request));
+    }
+    return event;
+  }
+
+  private static @NotNull User toUser(
+      final @Nullable User existingUser, final @NotNull HttpServletRequest request) {
+    final User user = existingUser != null ? existingUser : new User();
+    user.setIpAddress(toIpAddress(request));
```

Comment:
I'm afraid that this would require the `send_default_pii` flag to be added on options, which is disabled by default.
https://docs.sentry.io/error-reporting/configuration/?platform=python

Reply:
Seems like it is working for me

![image](https://user-images.githubusercontent.com/1357927/91168681-544ddc00-e6d6-11ea-9d9a-5e0c1e38f02e.png)

Reply:
but that's the problem, the SDK should only capture (automatically) PII if the flag `send_default_pii` is enabled, as we don't even have the flag, sending this info would be a PII problem, basically, to capture and send the IP address should only be possible if the user has enabled this flag like `sentry.sendDefaultPII=true`, as we don't even have/expose this flag yet, I'd say that we should skip the IP address for now (maybe just commenting the code and adding a TODO).
or to implement the flag on this or another PR.

Reply:
Ah ok, understood. I imagine username could be also considered PII. In case of Spring integration, this filter has to be added explicitly by the user as we are not able to autoconfigure it. Maybe it's ok then to stay as it's just an opt-in option?

Reply:
yeah usually those things are opt-in.
the thing is, we need to make it opt-in when we are able to collect this info automatically eg IP Address, if the user depends on the customer setting like Sentry.setUser(...) then it's fine, I'm just not sure if the IP address here is set by the customer or Spring offers this for free, that's the question that should be answered.

Reply:
We either take the ip address from the `X-FORWARDED-FOR` header set by load balancers or directly from `HttpServletRequest` object. But in order to use it, users have to explicitly create an instance of this filter and add it to Spring Security configuration (example https://github.com/getsentry/sentry-android/pull/517/files#diff-6c5b2fc2c37b9e5c8bdbdc86be9588ccR30)

Reply:
What @marandaneto said.
Could you please add `Boolean sendDefaultPii` to `SentryOptions` and checking for that before adding any of this user data by default?

Reply:
Ok, in this case it is enough to add it to `SentryProperties` but I am wondering if we shouldn't erase any personal information somewhere deeper in `sentry-core` if this flag is set to false?

Reply:
that's a good point, if the flag is a safeguard to not even set it, I don't see how we could do differently other than doing the check guard in all the setters/all the integrations.

the other option would be if `setIpAddress` gets a boolean to set it or not so at least we don't repeat the same `if` everywhere.

we could start doing this check guard in here and if we see the need for other places, we just refactor it?!

the other option would be, we always set it, if possible, and we could create a PiiEventProcessor that reads the flag and erase the values if set, it'd be an elegant way of solving it, but this allows the user to set it in the first place, I don't know if this is fine or not. cc @bruno-garcia

Reply:
We don't have any PII to be erased yet. We delayed adding the flag until the first one popped up.
Happy to merge this and add that later